core/src/main/scala/org/apache/spark/h2o/converters/H2OFrameFromRDDProductBuilder.scala

```diff
@@ -101,20 +101,28 @@ object H2OFrameFromRDDProductBuilder{
   private[converters] def perTypedDataPartition[T<:Product]()
                                                            (keyName: String, vecTypes: Array[Byte], uploadPlan: Option[immutable.Map[Int, NodeDesc]])
                                                            (context: TaskContext, it: Iterator[T]): (Int, Long) = {
+    val asArr = it.toArray[Product] // need to buffer the iterator in order to get its length
```

Comment:
Done 👍 Now it's done only for internal H2O backend and hopefully it's more readable